### ISSUE TITLE
[Feat Req] Option to  disable regex escaping of template variable value

### ISSUE THREAD

**Issue by corentingi** [grafana/grafana#7955]
- **Grafana 4.2**
- Datasource InfluxDB

Multiple choice variables are not replaced properly in a constant templating value.

### What I've done
I have multiple **InfluxDB query templatings** that I use in all my graphs.
Let's say I have a `$dc` and a `$host`. (I have actually 8 of those)
I created a **constant templating** with the following value:
**Name:** *filter* -- **Value:** `"host" =~ /^$host$/ AND "dc" =~ /^$dc$/`

The idea is to have `[...] WHERE $filter GROUP BY [...]` in my InfluxDB queries instead of putting the whole list every time.
(I get the same problem using *custom templating*)

### Expected result
The expected result would be to have this replacement:
`$filter -> "host" =~ /^.*$/ AND "dc" =~ /^dc-en|dc-fr$/`

### What happened instead
`$filter -> "host" =~ /^.*$/ AND "dc" =~ /^{dc-en,dc-fr}$/`

Basically, it replaces `.*` correctly.
But when replacing a list, it is not formated with the regular `item1|item2` regex but with a `{item1,item2}` list.

Maybe the constant templating was not designed to do this ?
I can see why a list is represented this way.
But when the constant is used inside a query I would expect it to be displayed the same as a regular query templating.

Thanks !

**Comment by corentingi** [comment]
Hum... As it often happens, I just discovered the Adhoc filters feature after writing this issue.
We can still discuss this issue, but I found a better solution to my use case 😃

**Comment by nheath** [comment]
👍 
We collect perfmon stats, including process details like memory, cpu, threads, etc, and write them to influxdb with an `instance` tag for the process name. I wanted to write the (unfortunately complicated) regex to match our many application processes, and reuse it across multiple graphs for the various measurements collected. Right now we are having to update that regex at least 20 times across our various graphs and dashboards. I'm not understanding the point of grafana allowing (and even suggesting) a regex filter of `/^$my_template_var$/` if regex isnt supported in variables.

**Comment by torkelo** [comment]
closing as similar to https://github.com/grafana/grafana/issues/6789
